LoveAndHiphop/LoveAndHiphop/MatchesViewController.swift

```diff
@@ -227,20 +221,142 @@ class MatchesViewController: UIViewController, UITableViewDelegate, UITableViewD
     
     print(" Yea me too")
     let detailViewController = segue.destination as! DetailViewController
+    
+    //Get all details of the current user
+    let curUserObjectId: String = (self.currentUserObject?.userObjectId)!
+    let curLikedByUsers: [String] = self.likedByUsersDict[curUserObjectId] ?? []
+    
+    //Get all details of the selected user
     let indexPath = self.tableView.indexPath(for: sender as! MatchCell)
-    let user = userObjects?[(indexPath?.row)!]
-    let userObjectId: String = (user?.userObjectId)!
-    detailViewController.likedByUsers = self.likedByUsersDict[userObjectId]
-    detailViewController.userObject = user
+    let selUser = userObjects?[(indexPath?.row)!]
+    let selUserObjectId: String = (selUser?.userObjectId)!
+    let selLikedByUsers: [String] = self.likedByUsersDict[selUserObjectId] ?? []
     
+    if(curLikedByUsers.contains(selUserObjectId) && selLikedByUsers.contains(curUserObjectId)) {
```

Comment:
This is the logic to check if two users like each other. 
cur - indicates the current user and sel - indicates the selected user (The cell that was selected). Here the logic is to see if the objectID of current user is in the list of LikedByUsers of the current users and viceversa. If so, we set the *isMatch* to true in DetailViewController.